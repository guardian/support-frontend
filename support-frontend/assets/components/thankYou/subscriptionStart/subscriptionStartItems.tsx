import { css } from '@emotion/react';
import { between, space } from '@guardian/source/foundations';
import type { ActiveProductKey } from 'helpers/productCatalog';

const downloadCopy = css`
	${between.desktop.and.leftCol} {
		max-width: 260px;
		display: block;
	}
`;
const subscriptionItems = css`
	& div:not(:first-child) {
		margin-top: ${space[4]}px;
	}
`;
const boldText = css`
	font-weight: bold;
`;
const paragraphSpacing = css`
	margin-bottom: ${space[1]}px;
`;

export const benefitsHeader = 'Whatâ€™s included?';
export function BenefitsBodyCopy(): JSX.Element {
	return (
		<span css={downloadCopy}>Your Digital + print subscription includes:</span>
	);
}
export const subscriptionStartHeader = 'When will your subscription start?';

type SubscriptionStartProps = {
	productKey: ActiveProductKey;
	startDate: string;
};
export function SubscriptionStartItems({
	productKey,
	startDate,
}: SubscriptionStartProps): JSX.Element | null {
	const paperCopy = (
		<span css={[downloadCopy, subscriptionItems]}>
			<div>
				<p>
					<span
						css={boldText}
					>{`You will receive your newspaper from ${startDate}`}</span>
				</p>
				{productKey === 'SubscriptionCard' && (
					<p css={paragraphSpacing}>
						You will receive your Subscription Card in your subscriber pack in
						the post, along with your home delivery letter.
					</p>
				)}
				<p>
					Visit your chosen participating newsagent to pick up your newspaper
					using your Subscription Card, or arrange a home delivery using your
					delivery letter.
				</p>
			</div>
		</span>
	);
	const guardianWeeklyCopy = (
		<span css={[downloadCopy, subscriptionItems]}>
			<div>
				<p>
					<span
						css={boldText}
					>{`Your first issue of Guardian Weekly will be published on ${startDate}`}</span>
				</p>
				<p>
					Please allow one to seven days after the publication date for your
					copy to be delivered to your door, depending on postal services.
				</p>
			</div>
		</span>
	);
	const copyContent: Partial<Record<ActiveProductKey, JSX.Element>> = {
		HomeDelivery: paperCopy,
		NationalDelivery: paperCopy,
		SubscriptionCard: paperCopy,
		GuardianWeeklyRestOfWorld: guardianWeeklyCopy,
		GuardianWeeklyDomestic: guardianWeeklyCopy,
		TierThree: (
			<span css={[downloadCopy, subscriptionItems]}>
				{guardianWeeklyCopy}
				<div>
					<p>
						<span css={boldText}>Your digital benefits start today.</span>
					</p>
					<p>
						Please ensure you are signed in on all your devices to enjoy all
						your benefits, including unlimited app access and uninterrupted
						ad-free reading.
					</p>
				</div>
			</span>
		),
	};
	return copyContent[productKey] ?? null;
}
