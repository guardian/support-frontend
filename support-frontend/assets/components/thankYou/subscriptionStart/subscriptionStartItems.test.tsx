import { render } from '@testing-library/react';
import type { ActiveProductKey } from 'helpers/productCatalog';
import { SubscriptionStartItems } from './subscriptionStartItems';

describe('Summary Ts&Cs Snapshot comparison', () => {
	const summaryProductKeys: ActiveProductKey[] = [
		'HomeDelivery',
		'NationalDelivery',
		'SubscriptionCard',
		'GuardianWeeklyRestOfWorld',
		'GuardianWeeklyDomestic',
		'TierThree',
	];
	it.each(summaryProductKeys)(
		`summaryTs&Cs render product %s correctly`,
		(summaryProductKey) => {
			const { container } = render(
				<SubscriptionStartItems
					productKey={summaryProductKey}
					startDate="01-01-25"
				/>,
			);
			expect(container).toMatchSnapshot();
		},
	);
});
