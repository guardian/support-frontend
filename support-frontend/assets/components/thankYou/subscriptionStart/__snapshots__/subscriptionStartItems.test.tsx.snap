// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product GuardianWeeklyDomestic correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <div>
      <p
        class="css-1378exm"
      >
        Your first issue of Guardian Weekly will be published on 01-01-25
      </p>
      <p>
        Please allow one to seven days after the publication date for your copy to be delivered to your door, depending on postal services.
      </p>
    </div>
  </span>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product GuardianWeeklyRestOfWorld correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <div>
      <p
        class="css-1378exm"
      >
        Your first issue of Guardian Weekly will be published on 01-01-25
      </p>
      <p>
        Please allow one to seven days after the publication date for your copy to be delivered to your door, depending on postal services.
      </p>
    </div>
  </span>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product HomeDelivery correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <div>
      <p
        class="css-1378exm"
      >
        You will receive your newspaper from 01-01-25
      </p>
      <p>
        Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.
      </p>
    </div>
  </span>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product NationalDelivery correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <div>
      <p
        class="css-1378exm"
      >
        You will receive your newspaper from 01-01-25
      </p>
      <p>
        Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.
      </p>
    </div>
  </span>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product SubscriptionCard correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <div>
      <p
        class="css-1378exm"
      >
        You will receive your newspaper from 01-01-25
      </p>
      <p
        class="css-c5d9qv"
      >
        You will receive your Subscription Card in your subscriber pack in the post, along with your home delivery letter.
      </p>
      <p>
        Visit your chosen participating newsagent to pick up your newspaper using your Subscription Card, or arrange a home delivery using your delivery letter.
      </p>
    </div>
  </span>
</div>
`;

exports[`Summary Ts&Cs Snapshot comparison summaryTs&Cs render product TierThree correctly 1`] = `
<div>
  <span
    class="css-1p5gon9"
  >
    <span
      class="css-1p5gon9"
    >
      <div>
        <p
          class="css-1378exm"
        >
          Your first issue of Guardian Weekly will be published on 01-01-25
        </p>
        <p>
          Please allow one to seven days after the publication date for your copy to be delivered to your door, depending on postal services.
        </p>
      </div>
    </span>
    <div>
      <p
        class="css-1378exm"
      >
        Your digital benefits start today.
      </p>
      <p>
        Please ensure you are signed in on all your devices to enjoy all your benefits, including unlimited app access and uninterrupted ad-free reading.
      </p>
    </div>
  </span>
</div>
`;
